### PR TITLE
Si/feat/model additions

### PR DESCRIPTION
**Pull request by siqbal181**
- Split out reply and comment schema and added timestamps

### CODE REVIEW

File: backend/src/models/postModel.ts [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+      required: true
+    },
+    avatar: {
+      type: String,
+      required: true
+    }
+  },
+  commentText: { type: String },
+  replies: [replySchema],
+  createdAt: { type: Date, default: Date.now },
```
Comment (pablisch) [inline backend/src/models/postModel.ts:40]: I saw a commit comment, I think, about the need to create the timestamp manually as timestamps only happen at the top level which makes sense. Great idea to separate these Schemas out but I was thinking that doing so would also mean that they are now at top level and could now use automatic timestamps? As before, not sure there is any great advantage but if it is possible then it feels like we should do so consistently.

File: frontend/src/components/PostItem/PostItem.tsx [hunk 2]
```diff
@@ -81,7 +95,11 @@ const PostItem: React.FC<PostObject> = ({
       <div>
         {images?.map((image, index) => (
           <div key={index}>
-            <img src={image.url} alt={image.caption} style={{ width: '300px' }} />
+            <img
+              src={image.url}
+              alt={image.caption}
+              style={{ width: "300px" }}
```
Comment (pablisch) [inline frontend/src/components/PostItem/PostItem.tsx:101]: I know this is not new but every time I see it, I wonder why this style is inline. Not something that needs to change for this PR but can we discuss inline styling vs stylesheets. I don't think we should mix the two as hierarchy is likely to come back and bite us on this.

**Review by pablisch — APPROVED** [review]
There's a couple of comments to read and do as you like but this is great

File: frontend/src/components/PostItem/PostItem.tsx (hunk 2 above)
Comment (siqbal181) [reply frontend/src/components/PostItem/PostItem.tsx:101]: Agree. This is all overhauled in the post PR which handles the new way the image looks anyways!

File: backend/src/models/postModel.ts (hunk 1 above)
Comment (siqbal181) [reply backend/src/models/postModel.ts:40]: Very true and obvious in hindsight!

**Review by pablisch — APPROVED** [review]
Noice!

**Pull request merged**